src/main/java/org/terasology/launcher/game/TerasologyGameVersions.java

```diff
@@ -83,6 +92,88 @@ public synchronized TerasologyGameVersion getGameVersionForBuildVersion(GameJob
         return null;
     }
 
+    public synchronized void loadGameVersionsFromPackageManager(GameSettings gameSettings, Path launcherDirectory, Path gameDirectory) {
```

Comment:
@praj-foss Please don't get confused, I like how you use stream API here to clean this code a bit up 👍 

Note to self: I'm not very happy with how this code mutates state, e.g., filling the version lists. One thing I (now) find difficult is to understand which part of the code actually changes which variables. 
At some point, I'd like to reduce the number of class members and group information logically together.